includes/CLI/Plugin_Check_Command.php

```diff
@@ -0,0 +1,423 @@
+				'line',
+				'column',
+				'type',
+				'code',
+				'message',
+			);
+		}
+
+		return new \WP_CLI\Formatter(
+			$assoc_args,
+			$default_fields
+		);
+	}
+
+	/**
+	 * Flattens and combines the given associative array of file errors and file warnings into a two-dimensional array.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $file_errors   Errors from a Check_Result, for a specific file.
+	 * @param array $file_warnings Warnings from a Check_Result, for a specific file.
+	 * @return array Combined file results.
+	 */
+	protected function flatten_file_results( $file_errors, $file_warnings ) {
+		$file_results = array();
+
+		foreach ( $file_errors as $line => $line_errors ) {
+			foreach ( $line_errors as $column => $column_errors ) {
+				foreach ( $column_errors as $column_error ) {
+
+					$file_results[] = array_merge(
+						$column_error,
+						array(
+							'type'   => 'ERROR',
+							'line'   => $line,
+							'column' => $column,
+						)
+					);
+				}
+			}
+		}
+
+		foreach ( $file_warnings as $line => $line_warnings ) {
+			foreach ( $line_warnings as $column => $column_warnings ) {
+				foreach ( $column_warnings as $column_warning ) {
+
+					$file_results[] = array_merge(
+						$column_warning,
+						array(
+							'type'   => 'WARNING',
+							'line'   => $line,
+							'column' => $column,
+						)
+					);
+				}
+			}
+		}
+
+		usort(
+			$file_results,
+			function( $a, $b ) {
+				if ( $a['line'] < $b['line'] ) {
+					return -1;
+				}
+				if ( $a['line'] > $b['line'] ) {
+					return 1;
+				}
+				if ( $a['column'] < $b['column'] ) {
+					return -1;
+				}
+				if ( $a['column'] > $b['column'] ) {
+					return 1;
+				}
+				return 0;
+			}
+		);
+
+		return $file_results;
+	}
+
+	/**
+	 * Display results.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param \WP_CLI\Formatter $formatter    Formatter class.
+	 * @param string            $file_name    File name.
+	 * @param array             $file_results Results.
+	 */
+	protected function display_results( $formatter, $file_name, $file_results ) {
+		\WP_CLI::line(
+			sprintf(
+				'FILE: %s',
+				$file_name
+			)
+		);
+
+		$formatter->display_items( $file_results );
+
+		\WP_CLI::line();
+		\WP_CLI::line();
+	}
+
+	/**
+	 * Converts a plugin basename back into a friendly slug.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $basename Plugin basename.
+	 * @return string Plugin slug.
+	 */
+	public function get_plugin_name( $basename ) {
+		if ( false === strpos( $basename, '/' ) ) {
+			$name = basename( $basename, '.php' );
+		} else {
+			$name = dirname( $basename );
+		}
+
+		return $name;
+	}
+
+	/**
+	 * Gets all available plugins.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * Uses the same filter core uses in plugins.php to determine which plugins
+	 * should be available to manage through the WP_Plugins_List_Table class.
+	 *
+	 * @return array
+	 */
+	private function get_all_plugins() {
+
+		return apply_filters( 'all_plugins', get_plugins() );
+	}
```

Comment:
See my above comment, those two methods won't be needed. Please keep in mind to not introduce public methods unnecessarily - especially here since it would mean that they then would become CLI commands, which we don't want.

Reply:
Removed it now, thanks!